Assemblers.Protocol/AssemblyFilter.cs

```diff
@@ -0,0 +1,185 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+using Skyline.DataMiner.CICD.Assemblers.Common;
+using NuGet.Packaging.Core;
+
+namespace Skyline.DataMiner.CICD.Assemblers.Protocol
+{
+    /// <summary>
+    /// Provides functionality to filter assemblies based on target framework, package references, and build result items.
+    /// </summary>
+    public static class AssemblyFilter
+    {
+        /// <summary>
+        /// Filters assemblies based on the specified criteria.
+        /// </summary>
+        /// <param name="targetFrameworkMoniker">The target framework moniker.</param>
+        /// <param name="packageReferenceProcessor">The package reference processor.</param>
+        /// <param name="buildResultItems">The build result items.</param>
+        /// <param name="dllImports">A set of DLL imports.</param>
+        /// <param name="packageIdentities">A list of package identities.</param>
+        /// <returns>The filtered NuGet package assembly data.</returns>
+        public static async Task<NuGetPackageAssemblyData> Filter(string targetFrameworkMoniker, PackageReferenceProcessor packageReferenceProcessor, BuildResultItems buildResultItems, HashSet<string> dllImports, IList<PackageIdentity> packageIdentities)
+        {
+            var nugetAssemblyData = await packageReferenceProcessor.ProcessAsync(packageIdentities, targetFrameworkMoniker,
+                Skyline.DataMiner.CICD.Common.NuGet.DevPackHelper.ProtocolDevPackNuGetDependenciesIncludingTransitive).ConfigureAwait(false);
+
+            ProcessFrameworkAssemblies(dllImports, nugetAssemblyData);
+            ProcessLibAssemblies(buildResultItems, dllImports, nugetAssemblyData);
+
+            return nugetAssemblyData;
+        }
+
+        /// <summary>
+        /// Adds a new DLL import to the set if it does not already contain an import with the same name.
+        /// </summary>
+        /// <param name="dllImports">A set of DLL imports.</param>
+        /// <param name="newImport">The new import to add.</param>
+        private static void AddToDllImport(HashSet<string> dllImports, string newImport)
```

Comment:
This will need to return a boolean to indicate wether it has been added.
The result then needs to be used in the calling site: e.g. if it has been added then an entry can be added to directoriesWithExplicitDllImport, otherwise it will need to be added to potentialRemainingDirectoryImports

Reply:
Applied. This was indeed needed to allow other dll's in the same nuget to still get found during runtime.By adding it to potentialRemainingDirectoryImports  it will be a directory path added to the dllIimports allowing it to be a location to look for assemblies during runtime.